### ISSUE TITLE
Feature: custom `!!` bangs/overrides

### ISSUE THREAD

**Issue by Silveere** [searxng/searxng#3055]
**Is your feature request related to a problem? Please describe.**

Currently, there doesn't seem to be any easy way to add custom external/double bangs.

**Describe the solution you'd like**

I'd like to see a configuration option added to `settings.yml` that allows for local overrides of the "external bangs" feature. Locally defined bangs should override ones retrieved from DuckDuckGo. Perhaps something like this:

<pre><code>search:
    bangs:
        'modrinth': 'https://modrinth.com/mods?q={query}' # create new bang
        'mcw': 'https://minecraft.wiki/?search={query}' # override existing bang
        'example': false # disable bang
</code></pre>

This example would make the query `!!modrinth fabric api` redirect to `https://modrinth.com/mods?q=fabric+api`, `!!mcw Creeper` redirect to `https://minecraft.wiki/?search=Creeper` (replacing the existing bang), and an existing `!!example` bang would be disabled as if it were never there.

**Describe alternatives you've considered**

Using custom browser search engines. However, this has to be manually configured on every browser I use; this is especially a pain on mobile and it is the one thing modern browsers do not sync for some reason.

**Additional context**

DuckDuckGo currently seems to have abandoned maintaining their bangs; new bangs haven't been added for a long time, and existing bangs are sometimes extremely outdated. For example, `!!mcw` goes to `https://minecraft.gamepedia.com`, which then redirects to `https://minecraft.fandom.com`, despite the fact that the "official" Minecraft wiki has moved off of Fandom in favor of `https://minecraft.wiki`; it has moved *twice* since it was last updated. Even if they were to start maintaining them again, DuckDuckGo also has some restrictions regarding bangs that they will accept; they will not accept bangs for adult websites and they must be "useful to more than around 500 people." <sup>[[source]](https://duckduckgo.com/newbang)</sup>

**Comment by glanham-jr** [comment]
Hmmmm, I think this might be tricky as entering a custom bang means being redirected to a third party site for searching, and the point of searxng is to be a meta search engine to avoid that entirely. Would it be better to add a feature engine request? Or do you use a lot of bangs? 

> Using custom browser search engines. However, this has to be manually configured on every browser I use; this is especially a pain on mobile and it is the one thing modern browsers do not sync for some reason.

There is FireFox sync if that is your thing, which should sync settings across browsers. But yes, mobile is especially a pain for these sort of things.

**Comment by Silveere** [comment]
Yes, bangs do indeed redirect the user to a third party site, but this is an already existing and established feature of SearXNG, and I do not see any issue with simply making it more configurable. I use bangs quite extensively, especially on search engines that only link to first-party resources (e.g., Wikipedia search). I've started to notice a few issues with DuckDuckGo's set of bangs that I've already pointed out in my original post under "Additional context", and I think SearXNG would benefit from making this feature configurable.

> There is FireFox sync if that is your thing, which should sync settings across browsers.

I have already tried Firefox sync, and custom search engines seem to be one of the few things that do not get synced.

**Comment by glanham-jr** [comment]
You are right - the double bang does exist. And it makes sense that if it exists, might as well be configurable. I don't mind working on this, but I'd prefer a maintainers opinion first so the PR is not rejected out of the gate.

> I have already tried Firefox sync, and custom search engines seem to be one of the few things that do not get synced.

Dang, what a let down. Seems like a very nice thing to sync, as these take time to create. Time to open an issue on Firefox Sync as well lol.

**Comment by xieve** [comment]
I can also imagine working on this. I use the double bang feature a lot, especially for less "search-like" websites like manpages, nixOS packages, shopping sites etc.
